src/JLD.jl

```diff
@@ -203,7 +203,7 @@ function jldopen(filename::AbstractString, rd::Bool, wr::Bool, cr::Bool, tr::Boo
             if sz < 512
                 error("File size indicates $filename cannot be a Julia data file")
             end
-            magic = Array(UInt8, 512)
+            magic = Array{UInt8}(512)
```

Comment:
best to also specify dimensionality `Array{UInt8,1}` or `Vector{UInt8}` when known